compiler/packages/babel-plugin-react-compiler/src/Inference/InferReferenceEffects.ts

```diff
@@ -521,7 +520,7 @@ class InferenceState {
          *   `expected valueKind to be 'Mutable' but found to be \`${valueKind}\``
          * );
          */
-        effect = isObjectType(place.identifier) ? Effect.Store : Effect.Mutate;
```

Comment:
This was always sketchy: why is only a Store effect on a known object type? A Store is a Store: you're either doing a transitive mutation or a local mutation. The conditional here was hiding incorrect code elsewhere, fixed in this PR.